detect.go

```diff
@@ -0,0 +1,106 @@
+package pip
+
+import (
+	"os"
+
+	"github.com/paketo-buildpacks/packit"
+)
+
+// BuildPlanMetadata is the buildpack specific data included in build plan
+// requirements.
+type BuildPlanMetadata struct {
+	// Build denotes the dependency is needed at build-time.
+	Build bool `toml:"build"`
+
+	// Launch denotes the dependency is needed at runtime.
+	Launch bool `toml:"launch"`
+
+	// Version denotes the version of a dependency, if there is one.
+	Version string `toml:"version"`
+
+	// VersionSource denotes where dependency version came from (e.g. an environment variable).
+	VersionSource string `toml:"version-source"`
+}
+
+// Detect will return a packit.DetectFunc that will be invoked during the
+// detect phase of the buildpack lifecycle.
+//
+// Detection always passes, and will contribute a  Build Plan that provides pip,
+// and requires cpython OR python, python_packages, and requirements.
+//
+// If a version is provided via the $BP_PIP_VERSION environment variable, that
+// version of pip will be a requirement.
+func Detect() packit.DetectFunc {
+	return func(context packit.DetectContext) (packit.DetectResult, error) {
+
+		requirements := []packit.BuildPlanRequirement{
+			{
+				Name: CPython,
+				Metadata: BuildPlanMetadata{
+					Build: true,
+				},
+			},
+		}
+
+		// TODO: remove this when removing legacy API
+		requirementsLegacy := []packit.BuildPlanRequirement{
+			{
+				Name: Python,
+				Metadata: BuildPlanMetadata{
+					Build:  true,
+					Launch: true,
+				},
+			},
+			{
+				Name: Dependency,
+				Metadata: BuildPlanMetadata{
```

Comment:
I haven't looked at the entire code but why is `python_packages` part of the API?
I don't see that in the issue, or think of any buildpack would be able to satisfy this requirement

Reply:
This issue contains `{python-runtime + requirements}` as part of the `OR`. I see that the [original buildpack detect](https://github.com/paketo-community/pip/blob/5f40d0df10625c0ebbbe241efac5cbd34a70d928/cmd/detect/main.go#L44-L55) requires these things, so in an effort to maintain the same behaviour we included them too

Reply:
But looks like the original buildpack provided and required `python_packages` as an internal contract. I believe the issue proposes that this contract be removed - I can't think of an external user of it. Is this serving any purpose you can think of?

Reply:
Ah I see now that `Dependency` is both provided and required in the original code, yes I think that can go -- you're right